Add search input for Directory list

DimaDevelopment opened the pull request:
There was a need to search the registry by name. I remember what it’s called, but scrolling through the whole list has become cumbersome due to the large number of items.

So I added an input that searches by registry name.
Overall, we could add animations and improve the search algorithm, making it more advanced — for example, searching by description as well.
But I feel that right now, it’s not really necessary.

Here’s how it works 👇

https://github.com/user-attachments/assets/081c21fe-eeb8-4e97-920e-8096c3d14b68

shadcn commented:
I did. Thanks for starting the work on it. Appreciate it. 

A few notes (if you have time):

1. Can we use nuqs? https://nuqs.dev/
2. Let's also add the description to the filter

DimaDevelopment commented:
@shadcn 
Look — I’ve added nuqs. I placed the NuqsProvider in the root layout so you can use it anywhere in the app later if needed.
Or do you think it’d be better to add it only to the documentation layout?
I also updated the search function to include descriptions — you can roll out a preview and check how it works!

shadcn commented on apps/v4/components/search-directory.tsx:
```diff
@@ -0,0 +1,45 @@
+            size="icon-xs"
+            onClick={() => setQuery(null)}
+          >
+            <XCircle />
```
Can we use https://lucide.dev/icons/x and only show it when we have input?

DimaDevelopment replied:
@shadcn, I’ll do it right now

shadcn reviewed (commented):
@DimaDevelopment This looks good. I left one minor comment. Can you take a look? I can merge it tonight.

franky47 commented on apps/v4/components/search-directory.tsx:
```diff
@@ -0,0 +1,45 @@
+            size="icon-xs"
+            onClick={() => setQuery(null)}
+          >
+            {query?.length > 0 && <X />}
```
**suggestion**: drop the optional chaining (query is non-nullable):
```suggestion
            {query.length > 0 && <X />}
```

shadcn reviewed (approved):
Perfect. Thank you.

franky47 commented:
Ah, I added one too 😅 #8756
